packages/playwright/src/common/esmLoaderHost.ts

```diff
@@ -21,21 +21,22 @@ import { PortTransport } from '../transform/portTransport';
 import { singleTSConfig, transformConfig } from '../transform/transform';
 
 let loaderChannel: PortTransport | undefined;
-// Node.js < 20
-if ((globalThis as any).__esmLoaderPortPreV20)
-  loaderChannel = createPortTransport((globalThis as any).__esmLoaderPortPreV20);
 
-// Node.js >= 20
-export let esmLoaderRegistered = false;
 export function registerESMLoader() {
+  // Opt-out switch.
+  if (process.env.PW_DISABLE_TS_ESM)
+    return;
+
+  if (loaderChannel)
+    return;
+
   const { port1, port2 } = new MessageChannel();
   // register will wait until the loader is initialized.
   require('node:module').register(url.pathToFileURL(require.resolve('../transform/esmLoader')), {
```

Comment:
Maybe throw a nice error if this api is not available (Node 16 or older), otherwise the clients could be puzzled ?

Reply:
Should we bump this one to 18 straight away? https://github.com/microsoft/playwright/blob/7da3be4a1a1213e94b8d9c11824805e58663cee7/packages/playwright-core/index.js

Reply:
Or that ^^. But IIUC most of the stuff still works on 16.

Reply:
Yeah, I don't think we should drop 16. Let me update this PR to make things better.